File: ACM-1/b.cpp
#include <iostream>
#include <cmath>
using namespace std;
bool tprime(int n){
    int cnt = 0;
    for(int i = 2; i < n/2 +1; i++ ){
        if(n % i == 0){
            cnt++;
        }
    }
    if(cnt == 1) return true;
    return false;
}
int main(){
    int n;
    cin>>n;
    int k;
    while(n--){
        cin>>k;
        if(tprime(k)){
            cout<<"YES"<<endl;
        }
        else{
            cout<<"NO"<<endl;
        }
    }

}

File: ACM-1/b.py
n = int(input())
cnt = 0
s = list()
while n > 0:
    k = int(input())
    for i in range(1, int(k**0.5) + 1):
        if k % i == 0:
            cnt+=1
    if cnt == 3:
        s.append("YES")
    else:
        s.append("NO")
    cnt = 0
    n = n - 1
print(s)